dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Protections/DistanceProtectionLineSide1.mo

```diff
@@ -0,0 +1,149 @@
+within Dynawo.Electrical.Controls.Protections;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model DistanceProtectionLineSide1 "Simple triangular distance relay, disconnects side 1 of the protected line when the apparent impedance in within a protected zone for longer than a set duration"
+  import Modelica.Constants;
+  import Dynawo.Connectors;
+  import Dynawo.NonElectrical.Logs.Timeline;
+  import Dynawo.NonElectrical.Logs.TimelineKeys;
+
+public
+  parameter Types.Time T1 "Time delay of trip in zone 1";
+  parameter Types.Time T2 "Time delay of trip in zone 2";
+  parameter Types.Time T3 "Time delay of trip in zone 3";
+  parameter Types.Time T4 "Time delay of trip in zone 4";
+
+  parameter Real R1Pu "Resistive reach of zone 1 in pu (base UNom, SNom)";
```

Comment:
Types.PerUnit could be used for all R and X

Reply:
base UNom, SNom or UNom, SnRef ? If it is for a line, I think it should be SnRef=100MVA.